handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -1536,12 +1570,30 @@ private void setHandshakeSuccess() {
         }
         ctx.fireUserEventTriggered(SslHandshakeCompletionEvent.SUCCESS);
 
+        if (TlsMasterKeyHandler.DoNothingTlsMasterKeyHandler.INSTANCE != masterKeyHandler) {
```

Comment:
@fzakaria after looking at the patch I think we should change it to just do it in a different `ChannelInboundHandler` and not adjust SslHandler at all.

Why not just do something like this:

```java
class TlsMasterKeyHandler extends ChannelInboundHandlerAdapter {

        @Override
        public void userEventTriggered(ChannelHandlerContext ctx, Object evt) {
            if (evt instanceof SslHandshakeCompletionEvent) {
                SslHandler handler = ctx.pipeline().get(SslHandler.class);
                SSLEngine engine = handler.engine();
                .....
               // Do all the logging here.
            }
            ctx.fireUserEventTriggered(evt);
        }
}
```

This way the user can just add the extra handler to the pipeline if interested in this... WDYT ?

Reply:
I think that is a very good suggestion. I will update the pull-request.